pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -858,27 +858,34 @@ func (p *PriorityQueue) podMatchesEvent(podInfo *framework.QueuedPodInfo, event
 		return true
 	}
 
-	// upon the same resource type, we need to check if the ActionType is *compatible* instead
-	// of *identical*.
 	for evt, nameSet := range p.clusterEventMap {
-		// Although rare, it's still allowed for plugins to return framework.WildCardEvent.
-		if evt == framework.WildCardEvent {
+		// Firstly verify if the two ClusterEvents match:
+		// - either the registered event from plugin side is a WildCardEvent,
+		// - or the two events have identical Resource fields and *compatible* ActionType.
+		//   Note the ActionTypes don't need to be *identical*. We check if the ANDed value
+		//   is zero or not. In this way, it's easy to tell Update&Delete is not compatible,
+		//   but Update&All is.
+		evtMatch := evt == framework.WildCardEvent ||
+			(evt.Resource == clusterEvent.Resource && evt.ActionType&clusterEvent.ActionType != 0)
+
+		// Secondly verify the plugin name matches.
+		// Note that if it doesn't match, we shouldn continue to search.
```

Comment:
It was my mistake to have suggested to directly return.

I don't think we need this comment. The code is self-explanatory.